internal/pkg/flink/internal/controller/statement_controller.go

```diff
@@ -100,3 +103,40 @@ func (c *StatementController) isCancelEvent() bool {
 	}
 	return false
 }
+
+func (c *StatementController) waitForStatementToBeInTerminalStateOrError(processedStatement types.ProcessedStatement) (*types.ProcessedStatement, *types.StatementError) {
+	readyStatementWithResults, err := c.store.FetchStatementResults(processedStatement)
+	if err != nil {
+		return nil, err
+	}
+
+	if readyStatementWithResults.IsTerminalState() {
+		return readyStatementWithResults, nil
+	}
+
+	ctx, cancelWaitForTerminalStatementState := context.WithCancel(context.Background())
+	defer cancelWaitForTerminalStatementState()
+
+	go c.listenForUserInputEvent(ctx, c.isDetachEvent, cancelWaitForTerminalStatementState)
+
+	output.Printf("Statement phase is %s. Press return to detach.\n", readyStatementWithResults.Status)
+	terminalStatement, err := c.store.WaitForTerminalStatementState(ctx, *readyStatementWithResults)
+	if err != nil {
+		return nil, err
+	}
+	return terminalStatement, nil
+}
+
+func (c *StatementController) isDetachEvent() bool {
+	if b, err := c.consoleParser.Read(); err == nil && len(b) > 0 {
+		pressedKey := prompt.Key(b[0])
+
+		switch pressedKey {
+		case prompt.ControlM:
+			fallthrough
+		case prompt.Enter:
```

Comment:
Not sure we need a `switch` if there's only one branch.
```suggestion
		case prompt.ControlM, prompt.Enter:
```